File: 2021/Day2/C1.py
class InstructionSet:
    def __init__(self):
        self.horizontal_pos = 0
        self.depth = 0
    def exec_forward(self,units):
        self.horizontal_pos+=units
    def exec_down(self,units):
        self.depth+=units
    def exec_up(self,units):
        self.depth-=units

    def solve_for(self, name: str,units:int):
        do = f"exec_{name}"
        if hasattr(self, do) and callable(func := getattr(self, do)):
            func(units)

def parseInstructions():
    instructionsHandler=InstructionSet()
    with open("input.txt") as instructions:
        instructions=instructions.readlines()
        for instruction in instructions:
            cmd,units=instruction.split(sep=" ")
            instructionsHandler.solve_for(cmd,int(units))
    return  instructionsHandler.depth*instructionsHandler.horizontal_pos


if __name__ == '__main__':
    print(parseInstructions())

File: 2021/Day2/C1.c
#include <stdio.h>
int compareString(char*cmd,char*instruction){
    int index=0;
    while(cmd[index]!='\0'){
        if(cmd[index]!=instruction[index]){
            return -1;
        }
        index++;
    }
    return 0;
}
int main()
{
    FILE *fileHandleIn = fopen("input.txt", "r");
    char buffer[20];
    int units=0;
    int horizontal=0;
    int depth=0;
    while ((fscanf(fileHandleIn, "%s", buffer)==1) && 
    (fscanf(fileHandleIn, "%d", &units)==1)){
       if(compareString("forward",buffer)==0){
           horizontal+=units;
       }else if (compareString("down",buffer)==0)
       {
           depth+=units;
       }else if (compareString("up",buffer)==0)
       {
           depth-=units;
       }
    }
    printf("%d",depth*horizontal);
}

